projects/element-ng/text-formatters/markdown-formatter.ts

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) Siemens 2016 - 2025
+ * SPDX-License-Identifier: MIT
+ */
+import { SecurityContext } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+
+import { ContentFormatter } from './types';
+
+/**
+ * Returns a markdown formatter function which_
+ * - Transforms markdown content into formatted HTML.
+ * - Returns a DOM node containing the formatted content.
+ *
+ * **Warning:** The returned Node is inserted without additional sanitization.
+ * Input content is sanitized before processing.
+ *
+ * @experimental
+ * @param sanitizer - Angular DomSanitizer instance
+ * @returns A function taking the markdown content to transform and returning a DOM div element containing the formatted HTML
+ */
+export const getMarkdownFormatter = (sanitizer: DomSanitizer): ((content: string) => Node) => {
```

Comment:
Not sure whether we should call this `renderer` or `formatter`. For me formatter sounds a bit more simpler, like formatting a number or a date string. This here feels more like a renderer to me, so I suggest to rename it but no strong opinion.